sdk/libs-scala/crypto/src/test/scala/com/daml/crypto/MessageDigestPrototypeSpec.scala

```diff
@@ -3,24 +3,32 @@
 
 package com.daml.crypto
 
+import org.bouncycastle.jce.provider.BouncyCastleProvider
+import org.scalatest.BeforeAndAfterAll
 import org.scalatest.flatspec.AnyFlatSpec
 import org.scalatest.matchers.should.Matchers
 
 import java.nio.charset.StandardCharsets
+import java.security.Security
 import java.util.Base64
 
-class MessageDigestPrototypeSpec extends AnyFlatSpec with Matchers {
+class MessageDigestPrototypeSpec extends AnyFlatSpec with Matchers with BeforeAndAfterAll {
 
   behavior of MessageDigestPrototype.getClass.getSimpleName
 
-  it should "provide new instance of digest for SHA-256" in {
-    val digest = MessageDigestPrototype.Sha256.newDigest
-    val digest2 = MessageDigestPrototype.Sha256.newDigest
-    digest should not be theSameInstanceAs(digest2)
+  override def beforeAll(): Unit = {
+    val _ = Security.insertProviderAt(new BouncyCastleProvider, 1)
```

Comment:
In order for `Secp256k1` curve to be available, we appear to need to force BouncyCastle to be at the start of the chain

Reply:
see comment below how you can get around that. you can also consider to add the provider in your MessageDigestPrototype and MessageSignaturePrototype objects to avoid adding BC as a provider in every test

Reply:
> consider to add the provider in your MessageDigestPrototype and MessageSignaturePrototype objects to avoid adding BC as a provider in every test

That's a good idea, thanks for that.